src/train_cmd.cpp

```diff
@@ -1662,19 +1662,31 @@ static bool TrainApproachingCrossing(TileIndex tile)
 	return HasVehicleOnPos(tile_from, &tile, &TrainApproachingCrossingEnum);
 }
 
+/** Check if the crossing should be closed
+ *  @return train on crossing || train approaching crossing || reserved
```

Comment:
We  normally don't use bit-logic in comments. Try to make it understandable for the average joe?